build.sh

```diff
@@ -6,33 +6,36 @@ else
   SED=sed
 fi
 
-stable() {
-  rustup toolchain add stable >&2
-  rustup run stable rustc --version | egrep -o '1\.[0-9]+\.[0-9]+'
+rustc_crawl() {
+  ptrn='\[pkg\.rust\]\nversion = "\K[^"]+'
+  relday=$(date -I)  # Today's date
+  # Rust stable updates happen every 6 weeks, starting at 2015-12-10.
+  # This would be a very clean method of extracting versions, were it not for patches.
+  # Rust 1.72.1 was released 21 days after 1.72.0 for example, but 1.73.0 was still
+  # on schedule. This makes last modified time of the stable file unreliable.
+  if [[ "$1" = 'stable' ]] ; then
+      rustup run stable rustc --version | grep -Eo '[0-9]+\.[0-9]+\.[0-9]+'
+      return 0
+  fi
+
+  # nightly releases should be found on the first check here, making it equivalent to an if,
+  # but beta releases are a pain in the behind to track down, so we probe a few days for those.
+  # On the off chance this ever runs before nightly builds, nightly falls back to yesterday's.
+  while [[ $(curl -s "https://static.rust-lang.org/?prefix=dist/$relday" | grep -c "$1") == '0' ]] ; do
+    relday=$(date -I -d "$relday -1 day")
+  done
+  # Extract whatever version information upstream provides!
+  curl -s "https://static.rust-lang.org/dist/$relday/channel-rust-$1.toml" | grep -ozP "$ptrn" | tr -d '\0'
 }
 
-beta() {
-  rustup toolchain add beta >&2
```

Comment:
The mastodon-poster relies on those being available, as it just runs `rustc --version` and posts that if it changes.

Reply:
Oversight on my end, will remove

Reply:
The `curl` call in `rustc_crawl` spits out the exact same info minus the `rustc ` prefix, so it should be possible to leverage a temp file which gets removed from the python end of things. Shaves off the `shlex` and `subprocess` calls as well. Would that work?